### PR TITLE
WIP: Add check workflow and fix tests

### PR DESCRIPTION
**Pull request by velezd**
- Unfortunately there isn't a way to fail a step and continue with execution or show a warning, so our extra tests are unskippable part of the check.
- "Proper fix" for our integration tests would require rewrite of the testing plugin. I though that is not necessary as this solution seems to be reliable enough.

### CODE REVIEW

File: .github/workflows/check.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    steps:
+      - name: Checkout change
+        uses: actions/checkout@v2
+        with:
+          path: permian
+
+      - name: Clone repository with additional plugins and tests
+        run: |
+          git  -c http.sslVerify=false clone --depth 1 ${{ secrets.ADDITIONAL_TESTS_REPO }} permian-redhat
```
Comment (pholica) [inline .github/workflows/check.yaml:24]: Shouldn't the variable be called something like secrets.ADDITIONAL_PLUGINS_REPO ?

**Review by pholica — COMMENTED** [review]
The changes in the test data look fine by me, I hope we won't be hitting any race condition in the test data again.

File: .github/workflows/check.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        run: |
+          pushd $GITHUB_WORKSPACE/permian
+          ./in_container make doc
+          popd
+
+      - name: Cleanup
+        if: always()
+        run: |
+          rm -rf $GITHUB_WORKSPACE/permian-redhat
+          podman rmi -f permian-redhat || true
```
Comment (pholica) [inline .github/workflows/check.yaml:43]: The permian-redhat seem to belong to the "base ("vanilla") cleanup, I guess this is just copy-paste remnant.

**Review by pholica — CHANGES_REQUESTED** [review]

File: .github/workflows/check.yaml [hunk 3]
```diff
@@ -0,0 +1,93 @@
+name: Check
```
Comment (jkonecny12) [inline .github/workflows/check.yaml:1]: I would suggest to use more specific name. `Check` is not telling much really.

File: .github/workflows/check.yaml [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    if: needs.secrets-check.outputs.extra_plugins_available == 'true'
+    steps:
+      - name: Checkout change
+        uses: actions/checkout@v2
+        with:
+          path: permian
+
+      - name: Clone repository with additional plugins and tests
+        run: |
+          git  -c http.sslVerify=false clone --depth 1 ${{ secrets.ADDITIONAL_TESTS_REPO }} permian-redhat
```
Comment (jkonecny12) [inline .github/workflows/check.yaml:67]: You have a double space here.

Comment (jkonecny12) [inline .github/workflows/check.yaml:67]: I would recommend to instead install the required certificate on our runners. They should already have some. Ping me of the PR if you want to go that way.

File: .github/workflows/check.yaml [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    steps:
+      - name: Check if ADDITIONAL_PLUGINS_REPO is set
+        id: extra_plugins
+        run: |
+            echo "::set-output name=extra_plugins_available::${{ secrets.ADDITIONAL_PLUGINS_REPO != '' }}"
+
+  extra-plugins:
+    needs: [secrets-check]
+    runs-on: [self-hosted, kstest]
+    if: needs.secrets-check.outputs.extra_plugins_available == 'true'
```
Comment (jkonecny12) [inline .github/workflows/check.yaml:58]: Is there a reason why not to do:
```
 if: secrets.ADDITIONAL_PLUGINS_REPO != ''
```

**Review by jkonecny12 — COMMENTED** [review]
I have some suggestions below.

File: .github/workflows/check.yaml (hunk 5 above)
Comment (velezd) [reply .github/workflows/check.yaml:58]: I tried it and it doesn't know secrets at that point. But I did it slightly differently so I can try again.

Comment (velezd) [reply .github/workflows/check.yaml]: Nope, doesn't work https://github.com/velezd/permian/actions/runs/1783600171

Comment (jkonecny12) [reply .github/workflows/check.yaml]: I worried it is this reason. I'll try to dig a bit about that...

File: .github/workflows/check.yaml (hunk 1 above)
Comment (velezd) [reply .github/workflows/check.yaml]: Yes, that is better name for it

File: .github/workflows/check.yaml (hunk 2 above)
Comment (velezd) [reply .github/workflows/check.yaml]: Fixed

**Review by pholica — APPROVED** [review]
Looks good. Thank you, I believe this should enable both downstream as well as upstream contributions/development.

**Comment by pholica** [comment]
Oh, it's still WIP, I also cannot see the Github actions result.

**Comment by jkonecny12** [comment]
Hi, you won't see result of the new action added by the same PR. You have to have it in the repository to be started on PRs.

**Comment by pholica** [comment]
Oh, thank you @jkonecny12 for the explanation. I cannot see that I'm happy about that but it makes sense now. Thank you. I'd just wait for @velezd to remove the WIP from this pull request before merging it.

**Pull request closed without merging**